text_reading/src/main/scala/org/clulab/aske/automates/apps/ExtractAndAlign.scala

```diff
@@ -123,6 +102,32 @@ object ExtractAndAlign {
     //    val variableNames = grfnVars.map(_.name.toUpperCase) // fixme: are all the variables uppercase?
 
     val grfn = ujson.read(grfnFile.readString())
+
+    //Getting comments from grfn
+    //the source_comments section of the json contains comments for multiple containers; get the container names to look up the comments for that container in the source_comments section
+    //todo: do we want to include "$file_head" and "$file_foot"?
+    val containerNames = grfn("containers").arr.map(_.obj("name").str)
+    val sourceCommentObject = grfn("source_comments").obj
+    //store comment strings here
+    val commentTexts = new ArrayBuffer[String]()
+    //for each container, the comment section has these three components
+    val commentComponents = List("head", "neck", "foot") //todo: a better way to read these in?
+    for (name <- containerNames) if (sourceCommentObject.contains(name)) {
+      val commentObject = sourceCommentObject(name).obj
+      for (cc <- commentComponents) if (commentObject.contains(cc)) {
+        val text = commentObject(cc).arr.map(_.str).mkString("")
+        if (text.length > 0) {
+          commentTexts.append(text)
+        }
+      }
+    }
+
+    // Parse the comment texts
+    val docs = commentTexts.map(parseCommentText(_, filename = Some(grfn("source").arr.head.str)))
```

Comment:
Since you're adding the filename here, let's use the doc.id down below where we make the link_elements.
Also -- should we add more context? i.e., should we append in the "head/neck/foot" and container name?

Reply:
the "down below is either line 187 or 195, IDK which number is your new one :)